subworkflows/local/assembly_qc.nf

```diff
@@ -54,18 +53,15 @@ workflow ASSEMBLY_QC {
 
         ch_versions = ch_versions.mix(BLAST_BLASTN_HOST.out.versions.first())
 
-        contaminated_contigs = Channel.of( BLAST_BLASTN_HUMAN_PHIX.out.txt, BLAST_BLASTN_HOST.out.txt )
-            .collectFile(name: "contaminated_contigs_host.txt", newLine: true)
+        contaminated_contigs = BLAST_BLASTN_HUMAN_PHIX.out.txt.mix( BLAST_BLASTN_HOST.out.txt )
     } else {
         contaminated_contigs = BLAST_BLASTN_HUMAN_PHIX.out.txt
     }
 
-    // TODO: this process only function is to publish the decontaminated contigs txt file
-    PUBLISH_DECONTAMINATED( contaminated_contigs )
-
+    // TODO: this is not fit for samplesheets kind of inputs
     SEQKIT_GREP(
         SEQKIT_SEQ.out.fastx,
-        contaminated_contigs.map { meta, hits_txt -> { hits_txt }}
+        contaminated_contigs.map { meta, hits_txt -> { hits_txt }}.collectFile(name: "contaminated_contigs.txt", newLine: true)
```

Comment:
Silly nextflow question - If both `params.remove_human_phix` and `host_reference_genome` are null, does this fail or simply not execute if `contaminated_contigs` is empty?

Reply:
Well. Indeed, this is going to be empty and we should return the contigs as they are. I'll fix this, good spot!

Reply:
Should be fixed in https://github.com/EBI-Metagenomics/miassembler/pull/10/commits/0bfa81c9517d5ee1c9df12dea32ce97d6e55e1bd